File: WitchcraftApp/src/main/java/com/tarbadev/witchcraft/recipes/domain/usecase/DeleteRecipeUseCase.java
package com.tarbadev.witchcraft.recipes.domain.usecase;

import com.tarbadev.witchcraft.recipes.domain.repository.RecipeRepository;
import org.springframework.stereotype.Component;

@Component
public class DeleteRecipeUseCase {
  private final RecipeRepository recipeRepository;

  public DeleteRecipeUseCase(RecipeRepository recipeRepository) {
    this.recipeRepository = recipeRepository;
  }

  public void execute(int id) {
    recipeRepository.delete(id);
  }
}

File: WitchcraftApp/src/main/java/com/tarbadev/witchcraft/recipes/domain/usecase/DeleteRecipeUseCase.kt
package com.tarbadev.witchcraft.recipes.domain.usecase

import com.tarbadev.witchcraft.recipes.domain.repository.RecipeRepository
import org.springframework.stereotype.Component

@Component
class DeleteRecipeUseCase(private val recipeRepository: RecipeRepository) {

    fun execute(id: Int) {
        recipeRepository.delete(id)
    }
}
